phoenix-pherf/src/main/java/org/apache/phoenix/pherf/workload/MultiThreadedRunner.java

```diff
@@ -88,13 +90,21 @@
      */
     @Override
     public Void call() throws Exception {
-        LOGGER.info("\n\nThread Starting " + threadName + " ; " + query.getStatement() + " for "
-                + numberOfExecutions + "times\n\n");
+        LOGGER.info("\n\nThread Starting " + threadName + " ; '" + query.getStatement() + "' for "
+                + numberOfExecutions + " times\n\n");
         Long start = EnvironmentEdgeManager.currentTimeMillis();
-        for (long i = numberOfExecutions; (i > 0 && ((EnvironmentEdgeManager.currentTimeMillis() - start)
-                < executionDurationInMs)); i--) {
+        for (long i = 0; i < numberOfExecutions; i++) {
+
+            if ((EnvironmentEdgeManager.currentTimeMillis() - start) >= executionDurationInMs) {
+                LOGGER.info("Queryset timeout of " + executionDurationInMs + " ms reached; current time is " + ((EnvironmentEdgeManager.currentTimeMillis() - start)) + " ms."
```

Comment:
There's a change in the logline as well as moving the logic for breaking the loop in case of executionDuration being exceeded
It was in the for loop definition before, now it's an if check at the beginning of the loop